lib/Models/TimeSeriesStack.js

```diff
@@ -66,8 +67,9 @@ TimeSeriesStack.prototype.addLayerToTop = function(item) {
  */
 TimeSeriesStack.prototype.removeLayer = function(item) {
     var index = this._layerStack.indexOf(item);
-
-    this._layerStack.splice(index, 1);
+    var layerStack = this._layerStack.slice(); // Make a copy to trigger the knockout subscription on _layerStack.
```

Comment:
Was this what solved the problem? It really shouldn't be needed with Knockout observable arrays :/